### PR TITLE
Added inverse hyperbolic functions (#261)

### PR DESCRIPTION
**Pull request by Jollywatt**
* Added `asinh`, `acosh`, `atanh` with tests. This completes support for
Julia's suite of standard trig functions.

* Commented-out a seemingly moot test asserting that the number
of `_dict_binary_ops` entries be `22`.

* Minor typo in docs.

### CODE REVIEW

**Comment by Jollywatt** [comment]
(I see "Some checks were not successful," but not sure the failed check is related to this PR. When I view the details, it reports that the `documenter/deploy` job succeeded in 3m 20s.)

File: src/dictmutfunct.jl [hunk 1]
```diff
@@ -115,6 +115,12 @@ const _dict_unary_ops = Dict(
         :(_aux = sinh(_arg1))],
     :tanh => [:tanh!, (:_res, :_arg1, :_aux, :_k), :(_res = tanh(_arg1)),
         :(_aux = tanh(_arg1)^2)],
+    :asinh => [:asin!, (:_res, :_arg1, :_aux, :_k), :(_res = asinh(_arg1)),
+        :(_aux = sqrt(_arg1^2 + 1))],
+    :acosh => [:acos!, (:_res, :_arg1, :_aux, :_k), :(_res = acosh(_arg1)),
+        :(_aux = sqrt(_arg1^2 - 1))],
+    :atanh => [:atan!, (:_res, :_arg1, :_aux, :_k), :(_res = atanh(_arg1)),
+        :(_aux = 1 - _arg1^2)],
```
Comment (lbenet) [inline src/dictmutfunct.jl:123]: The first argument in these functions is the name (as a symbol) of the (internal) function that deals with them. So they should be `:asinh!`, `:acosh!` and `:atanh!`, respectively. (My apologies for the lack of proper docs for this dictionaries.)

**Review by lbenet — COMMENTED** [review]
Thanks for this PR, and sorry for the long delay to get into it. I only spotted one simple typo (in `src/dictmutfunct.jl`) which should be very simple to correct. It would also be nice if you could add come docstrings (at the end of `src/functions.jl`). Aside from that, LGTM!

**Comment by lbenet** [comment]
> (I see "Some checks were not successful," but not sure the failed check is related to this PR. When I view the details, it reports that the documenter/deploy job succeeded in 3m 20s.)

I think that error is related to the deployment of the docs, which somehow does not work when the PR comes from a fork.

**Comment by Jollywatt** [comment]
@lbenet In addition to the typos and docs you suggested, I made the `ArgumentError`s thrown by the inverse trig functions more informative (see last commit). E.g.:
```julia
julia> asin(Taylor1([1, 0]))
ERROR: ArgumentError: Series expansion of asin(x) diverges at x = ±1
```

(Should these be `DomainError`s instead, mirroring the way Julia handles, e.g.,
```julia
julia> asin(2)
ERROR: DomainError with 2.0:
asin(x) is not defined for |x|>1.
```
? The rest of `TaylorSeries` uses `ArgumentError`s.)

**Comment by lbenet** [comment]
Thanks for considering my suggestions for this PR! LGTM!

> Should these be DomainErrors instead, mirroring the way Julia handles

Good point! Indeed, we throw `ArgumentError`s when we should be throwing `DomainError`s. Would you like to change that in this PR, or would you prefer doing it in a new one? (This issue also applies to other functions, such as `sqrt` or `log`.)

**Comment by Jollywatt** [comment]
I think I better open a new PR to do that, after this one is merged!

Jo

**Review by lbenet — APPROVED** [review]
LGTM! I'll go ahead and merge this! 

Thanks a lot

**Pull request merged**